xla/service/gpu/runtime/nccl_collective_permute_thunk.cc

```diff
@@ -53,6 +53,31 @@ absl::StatusOr<const int64_t> GetCurrentId(
           : current_logical_id.computation_id;
   return current_id;
 }
+
+bool IsLocalPeerTransfer(
+    const NcclP2PConfig::SourceTargetMapEntry& source_target,
+    se::Stream& stream, const int64_t current_id, const int64_t device_count) {
+  const std::optional<int64_t> source_id = source_target.source;
+  const std::optional<int64_t> target_id = source_target.target;
+  // Since mixing nccl p2p with p2p memcopy will cause random deadlocks.
```

Comment:
Can you elaborate where you see deadlocks? Will this also happen if some nodes decide that their permute is within a host and some other nodes decide that their permute is across hosts?

Reply:
the deadlocks could happen when calling nccl call and cuda memcpy p2p together(which both are synchronizing devices), in this case if this rank is sending across host using a nccl call but receiving from a local peer which is going through cuda api, the deadlock could happen because nccl cannot ensure the order of cuda api calls.

Reply:
added